monai/apps/auto3dseg/data_analyzer.py

```diff
@@ -259,17 +267,29 @@ def get_all_case_stats(self, key="training", transform_list=None):
                         DataStatsKeys.LABEL_STATS: d[DataStatsKeys.LABEL_STATS],
                     }
                 )
-            result[DataStatsKeys.BY_CASE].append(stats_by_cases)
+            result_bycase[DataStatsKeys.BY_CASE].append(stats_by_cases)
+
+        n_cases = len(result_bycase[DataStatsKeys.BY_CASE])
 
-        result[DataStatsKeys.SUMMARY] = summarizer.summarize(cast(list, result[DataStatsKeys.BY_CASE]))
+        result[DataStatsKeys.SUMMARY] = summarizer.summarize(cast(list, result_bycase[DataStatsKeys.BY_CASE]))
+        result[DataStatsKeys.SUMMARY]["n_cases"] = n_cases
+        result[DataStatsKeys.BY_CASE] = [None] * n_cases
```

Comment:
this is not compatible with the FL use cases https://github.com/Project-MONAI/MONAI/blob/d118aa7ac61ba23c7b8e83ba5558a72d82b2d0a2/monai/fl/client/monai_algo.py#L273-L275

Reply:
So I can update fl client to read 2 files. 

Or I can make it as an option in DataAnalyzer to save 1 file or 2 files. 

Which one is better ? But we do need 2 files in Auto3dseg to speed things up.

Reply:
I think we should somehow still return the by_case results in line 301, the fl client is making use of the returned value of `get_all_case_stats` https://github.com/Project-MONAI/MONAI/blob/d118aa7ac61ba23c7b8e83ba5558a72d82b2d0a2/monai/fl/client/monai_algo.py#L273

Reply:
@wyli  I've updated the code to "return" the combined result as before. plz check

Reply:
thanks, the test `python -m tests.test_fl_monai_algo_stats`  works fine now, I'm merging this, please make sure you still get the expected speed up...

Reply:
so for some reason my last change wasn't reflected on this branch, so I just removed an empty line and pushed again. now the test tests.test_fl_monai_algo_stats  passes ok. and it should be ok now